dataset/dataset.cpp

```diff
@@ -175,6 +148,26 @@ Dataset Dataset::slice(const Slice s) const {
   return out;
 }
 
+Dataset &Dataset::setSlice(const Slice s, const Dataset &data) {
+  // TODO Need dry-run mechanism here?
```

Comment:
Yes. However, for readonly Variables, this needs to perform the operation and compare the result to the currently stored values, making the dry run expensive.

Reply:
So we need several different checking functions?

Reply:
Can you elaborate?

Reply:
Here, we need a function that checks the dimensions and readonly flag. But in other places we need this in addition to the no-op check with the values.

Reply:
Well, we basically need a dry-run variant of `setSlice`.